R/modify-records.R

```diff
@@ -0,0 +1,151 @@
+#' Modify options and records from a `NONMEM` control stream file
+#' @param .mod a bbr model object
+#' @name modify_records
+#'
+#' @details
+#'  - **`safe_read_ctl()`** is called internally within the other functions, though
+#'  it can also be used outside of that context.
+#'  - **`modify_prob_statement()`**, **`modify_data_path_ctl()`**, and
+#'   **`remove_records()`** read in the control stream, make any modifications,
+#'   and then save out the updated control stream.
+#'     - `modify_prob_statement()` also returns a character string defining the
+#'     `$PROBLEM` text (see `prob_text` argument).
+#' - **`mod_has_record()`** will return a logical value denoting whether a `bbr`
+#'   model has a given record type.
+NULL
+
+#' @describeIn modify_records Safely read in a `NONMEM` control stream file via
+#' `nmrec`
+#' @keywords internal
+safe_read_ctl <- function(.mod){
+  check_model_object(.mod, c(NM_MOD_CLASS, NMBOOT_MOD_CLASS))
+  mod_path <- get_model_path(.mod)
+  ctl <- tryCatch(nmrec::read_ctl(mod_path), error = identity)
+  if(inherits(ctl, "error")){
+    warning(
+      glue::glue("Could not read control stream file. Reason: {ctl$message}")
+    )
+  }
```

Comment:
Addressed in [4f51be1](https://github.com/metrumresearchgroup/bbr/pull/687/commits/4f51be1c76d8260170288b29f9fa9b90cc082987)

 - It's now just a convenience wrapper for getting the control stream from a model object (rather than the model path)